pkg/trace/api/otlp.go

```diff
@@ -147,7 +146,7 @@ func (o *OTLPReceiver) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		}
 		r = gzipr
 	}
-	rd := apiutil.NewLimitedReader(r, o.conf.OTLPReceiver.MaxRequestBytes)
+	rd := io.LimitReader(r, o.conf.OTLPReceiver.MaxRequestBytes)
```

Comment:
I don't think the two are interchangeable. I don't remember the exact difference, but I do remember trying to do this myself and ditch `apiutil` in the past, but have found a notable difference.

Instead, how about completely removing the HTTP server which is not used?

Reply:
I think it can be replaced on this particular instance but not in other parts of the trace Agent. But you are right, let's just remove the HTTP server.